source/byte_buf.c

```diff
@@ -257,6 +257,44 @@ int aws_byte_cursor_split_on_char(
     return aws_byte_cursor_split_on_char_n(input_str, split_on, 0, output);
 }
 
+int aws_byte_cursor_find_exact(
+    const struct aws_byte_cursor *AWS_RESTRICT input_str,
+    const struct aws_byte_cursor *AWS_RESTRICT to_find,
```

Comment:
nitpick: I don't think those inputs need to be marked restrict, because you don't read from one after you write to the other. So, there's no performance penalty if they alias.

That being said, I don't think it hurts.

Reply:
I can't promise it won't be when I finally come back and implement Boyer-Moore